### PR TITLE
Add example on using abstract types for color

### PR DESCRIPTION
**Pull request by dstrekelj**
This example demonstrates how color can be abstracted over an `Int` type. Nothing new, but it should be a fun example for newcomers.

### CODE REVIEW

**Comment by Simn** [comment]
Very nice!

Could you add a short example of what the generated JavaScript code looks like? If we don't show that the point of using an abstract doesn't really come across.

**Comment by dstrekelj** [comment]
@Simn: Believe it or not, I was actually thinking about whether to include compiler output or not. Ultimately decided against it because I wasn't sure what compiler configuration would be most appropriate (e.g. full DCE with static analyzer enabled, or something else).

I guess I'll use that since it's the default on try.haxe.

File: assets/content/cookbook/Abstract types/color.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+```
+## Usage
+
+```haxe
+// Use a predefined color
+var red = Color.RED;
+// Use a custom color
+var fromInt = 0x98765432;
+var fromString = "0xffeeddcc";
```
Comment (markknol) [inline assets/content/cookbook/Abstract types/color.md]: I think you need to define the type (`var var fromString:Color = "0xffeeddcc";` in order to make this work?

Comment (dstrekelj) [inline assets/content/cookbook/Abstract types/color.md]: Haha, yeah, I just noticed I forgot to cast those variables. I'll fix it when I add in compiler outputs. Thanks for bringing it up!

**Comment by dstrekelj** [comment]
@Simn Is this alright?

**Comment by Simn** [comment]
Do we really have to add the shitty 3.2.1 output? :D

**Comment by dstrekelj** [comment]
I don't know, do we? :D I thought I'd leave it in because it's the latest stable version of Haxe. Of course, it's not a problem to remove it.

**Comment by Simn** [comment]
Let's just show the current output then.

**Comment by Simn** [comment]
Thanks! This is a really high-quality example, keep it up!

**Pull request merged**